### PR TITLE
Add 'option.filterRe' & '--filter' (CLI) to restrict download URLs via regexp string

### PR DESCRIPTION
**Pull request by puzrin**
Needed for https://github.com/rxaviers/cldr-data-npm/issues/44

Quick test:

```sh
./bin/download.sh -i urls.json -o ./tmp -f --filter '(core|numbers)'
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -68,6 +68,14 @@ module.exports = function(srcUrl, destPath, options, callback) {
   // Download
   }).then(function() {
     var srcUrls = alwaysArray(srcUrl);
+
+    if (options.filter) {
+      var filter_re = options.filter;
```
Comment (rxaviers) [inline index.js]: nitpick review: all variables use camelCase, let's rename this one `filterRe`.

File: index.js [hunk 2]
```diff
@@ -68,6 +68,14 @@ module.exports = function(srcUrl, destPath, options, callback) {
...
+      if (typeof filter_re === 'string') filter_re = new RegExp(filter_re);
```
Comment (rxaviers) [inline index.js]: Another nitpick style one: let's not inline and use double quotes, i.e.:

```diff
- if (typeof filter_re === 'string') filter_re = new RegExp(filter_re);
+ if (typeof filter_re === "string") {
+     filter_re = new RegExp(filter_re);
+ }
```

File: index.js [hunk 3]
```diff
@@ -68,6 +68,14 @@ module.exports = function(srcUrl, destPath, options, callback) {
...
+
+      srcUrls = srcUrls.filter(function(url) { return filter_re.test(url); });
```
Comment (rxaviers) [inline index.js]:
```diff
- srcUrls = srcUrls.filter(function(url) { return filter_re.test(url); });
+ srcUrls = srcUrls.filter(function(url) {
+     return filter_re.test(url);
+ });
```

**Comment by rxaviers** [comment]
This also needs documentation update README.md and proper handling via package.json equivalent property.

**Comment by rxaviers** [comment]
> This also needs documentation update README.md and proper handling via package.json equivalent property.

Nevermind! Not for this repo.

**Comment by puzrin** [comment]
Updated.

**Comment by rxaviers** [comment]
Thanks

**Comment by rxaviers** [comment]
Released as 0.3.0

**Comment by puzrin** [comment]
Seems not published https://www.npmjs.com/package/cldr-data-downloader still 0.2.5 there.

**Comment by rxaviers** [comment]
Oops, now it is, sorry.

**Pull request merged**